src/core/nearfieldreader.cpp

```diff
@@ -0,0 +1,144 @@
+ begin                : 27.08.2023
+ copyright            : (C) 2023 by Mathieu Pellerin
+ email                : mathieu (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "nearfieldreader.h"
+
+#include <QDebug>
+#include <QNdefMessage>
+#include <QNdefNfcTextRecord>
+#include <QNearFieldTarget>
+#include <QUrl>
+#include <qgsmessagelog.h>
+
+NearFieldReader::NearFieldReader( QObject *parent )
+  : QObject( parent )
+{
+  mNearFieldManager = std::make_unique<QNearFieldManager>( this );
+  connect( mNearFieldManager.get(), &QNearFieldManager::targetDetected, this, &NearFieldReader::handleTargetDetected );
+  connect( mNearFieldManager.get(), &QNearFieldManager::targetDetected, this, &NearFieldReader::handleTargetLost );
+}
+
+NearFieldReader::~NearFieldReader()
+{
+#if QT_VERSION < QT_VERSION_CHECK( 6, 0, 0 )
+  mNearFieldManager->setTargetAccessModes( QNearFieldManager::NoTargetAccess );
+#endif
+  mNearFieldManager->stopTargetDetection();
+}
+
+void NearFieldReader::handleTargetDetected( QNearFieldTarget *target )
+{
+  connect( target, &QNearFieldTarget::ndefMessageRead, this, &NearFieldReader::handleNdefMessageRead );
+  connect( target, &QNearFieldTarget::error, this, &NearFieldReader::handleTargetError );
+
+  emit targetDetected( QString( target->uid() ) );
+
+  if ( target->hasNdefMessage() )
+  {
+    mReadString.clear();
+    QNearFieldTarget::RequestId request = target->readNdefMessages();
+    if ( !request.isValid() )
+    {
+      handleTargetError( QNearFieldTarget::NdefReadError, request );
+    }
+  }
+}
+
+void NearFieldReader::handleTargetLost( QNearFieldTarget *target )
+{
+  disconnect( target );
+}
+
+void NearFieldReader::handleNdefMessageRead( const QNdefMessage &message )
+{
+  qInfo() << QStringLiteral( "Received %1 record message(s) from near-field target" ).arg( message.size() );
+  for ( const QNdefRecord &record : message )
+  {
+    switch ( record.typeNameFormat() )
+    {
+      case QNdefRecord::NfcRtd:
+      {
+        if ( record.isRecordType<QNdefNfcTextRecord>() )
+        {
+          QNdefNfcTextRecord textRecord( record );
+          mReadString.append( textRecord.text() );
+        }
+        else
+        {
+          mReadString.append( record.payload() );
+        }
+        emit readStringChanged();
+        break;
+      }
+
+      case QNdefRecord::Mime:
+      case QNdefRecord::Uri:
```

Comment:
Can this be converted to text too?

Reply:
@m-kuhn , it can be converted, however, I'm a bit fearful of appending it to the decoded text for now, I'm worried there are distinct uses for these things that might interfere with the goal of fetching text from tag. Thoughts?